src/plugins/data_source_management/public/auth_registry/authentication_methods_registry.ts

```diff
@@ -9,7 +9,7 @@ import { EuiSuperSelectOption } from '@elastic/eui';
 export interface AuthenticationMethod {
   name: string;
   credentialSourceOption: EuiSuperSelectOption<string>;
-  credentialForm?: React.JSX.Element;
+  credentialForm?: (state?: any, setState?: any) => React.JSX.Element;
```

Comment:
Let me set "state" and "setState" as required instead of optional because registered credential form need those parameters to interact with parent view.

For the strong typing of state and setState, Let me update the type to:

` credentialForm?: (state: { [key: string]: any }, setState: React.Dispatch<React.SetStateAction<any>>) => React.JSX.Element;`

Since the dataSourceManagement plugin should not rely on pre-existing knowledge about registered AuthTypes from other plugins, we still need any in value field.